### PR TITLE
bump_and_tag

### PR DESCRIPTION
**Pull request by dubchuck**
## Ticket
https://app.asana.com/0/1201247707079024/1202833279483342

## Goal
Migrate bump_and_tag to shared repo

## Code Changes
Add bump and tag

## How I know it works
https://github.com/HausAnalytics/api/runs/8046240183?check_suite_focus=true

### Checks
- [X] Changes in this PR follows Haus' [code contribution guidelines](https://docs.google.com/document/d/1-3pOkN0Qk5OrsoDJCFPL_Jwunc6ptXLzSpTdTRAH7to/edit#heading=h.fy8j7l7xqyx8) or if it breaks guidelines I've explained why it's necessary.
- [ ] I'm making a vmajor or vminor [semver](https://semver.org/) change and have that present in the branch name

### CODE REVIEW

**Comment by drorasaf** [comment]
why delete the test workflow to post to slack?

**Comment by dubchuck** [comment]
I think it was a relic of the merge issues last week.  I put it back in.

File: .github/workflows/bump_and_tag_with_google_creds.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      github_token_secret_name:
+        required: true
+        type: string      
+jobs:
+  bump-and-tag:
+    timeout-minutes: 3
+    runs-on: ubuntu-latest
+    steps:
+    - name: Auth with GCP
+      uses: 'google-github-actions/auth@v0.8.0'
```
Comment (drorasaf) [inline .github/workflows/bump_and_tag_with_google_creds.yml:24]: inconsistency if we need quotes or not for github actions

File: .github/workflows/bump_and_tag_with_google_creds.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Bump and tag with Google Credentials
+permissions:
```
Comment (drorasaf) [inline .github/workflows/bump_and_tag_with_google_creds.yml:2]: is this needed or does the caller need to have it? or both?

Comment (dubchuck) [reply .github/workflows/bump_and_tag_with_google_creds.yml:2]: The caller will need contents: read, but not id_token: write.  Updated

**Review by drorasaf — APPROVED** [review]

**Pull request merged**